tests/test_chat.py

```diff
@@ -0,0 +1,84 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from autogpt.agent import Agent
+from autogpt.config.ai_config import AIConfig
+from autogpt.llm.chat import chat_with_ai
+
+
+@pytest.fixture
+def agent():
+    ai_name = "Test AI"
+    memory = MagicMock()
+    full_message_history = []
+    next_action_count = 0
+    command_registry = MagicMock()
+    config = AIConfig()
+    system_prompt = "System prompt"
+    triggering_prompt = "Triggering prompt"
+    workspace_directory = "workspace_directory"
+
+    agent = Agent(
+        ai_name,
+        memory,
+        full_message_history,
+        next_action_count,
+        command_registry,
+        config,
+        system_prompt,
+        triggering_prompt,
+        workspace_directory,
+    )
+    return agent
+
+
+def test_chat_with_ai_basic_response(mocker, agent):
+    prompt = "Welcome to the autogpt!"
+    user_input = "Hello, how are you?"
+    full_message_history = []
+    permanent_memory = agent.memory
+    token_limit = 4000
+    mocker.patch(
+        "autogpt.llm.chat.create_chat_completion",
+        return_value="I'm doing well, thank you for asking.",
```

Comment:
im still learning mroe about mocker so excuse me if im incorrect, but by doing this `return value` and then asserting it arent we always going to get a passing test because the way you are asserting it is just going to return this? please feel free to correct me if im wrong

Reply:
Since the patched module is create_chat_completion it might be a correct implementation, so let me know if I'm missing something